### DISCUSSION TITLE
Unexpected Behavior in Boot Slot Switching: BOOT_ORDER Remains Unchanged in uboot.sh

### DISCUSSION THREAD

**Discussion by Jierger** [rauc/rauc#1439] [Q&A]
Hi All,

I am currently trying to port RAUC and I have a question regarding BOOT_ORDER when using uboot.sh,

I've noticed that if it's not marked, the system eventually switches boot slots, but the BOOT_ORDER doesn't change accordingly. It is expected to change from (A B) to (B A), but it remains (A B) to (A B). Is this the expected behavior? At this point, the marked order still doesn't change. Although I am not running this on a Raspberry Pi, I would like to confirm if you are experiencing the same behavior.

![image](https://github.com/rauc/rauc/assets/8401549/e93ef22a-e361-420a-a4cd-f4e17fc2b1ac)

**Comment by ejoerns** [comment]
@Jierger With `mark good`, RAUC currently does not modify the boot order in any way, yes.
This should be consistent with what's documented: https://rauc.readthedocs.io/en/latest/reference.html#bootloader-interaction

What changes the boot order is `mark active` only.

The assumption so far is that even if it failed booting, the boot slot marked as 'active' is still the 'preferred' one (but failed for some reason and is temporarily or persistently deactivated).
